model/src/main/scala/ch.epfl.scala.index.model/Release.scala

```diff
@@ -51,22 +52,44 @@ case class Release(
       Some(s"https://www.javadoc.io/doc/$groupId/$artifactId/$version")
     } else None
   }
+
+  lazy val orderedDependencies = scalaDependencies.sortBy(_.scope.contains(Scope.Test))
+  lazy val orderedJavaDependencies = javaDependencies.sortBy(_.scope.contains(Scope.Test))
+  lazy val dependencyCount = scalaDependencies.size + javaDependencies.size
+
+}
+
+sealed trait GeneralReference {
+  def groupId: String
+  def artifactId: String
+
+  def name: String
+  def httpUrl: String
 }
+
 object Release{
   case class Reference(
-    organization: String,     // typelevel               | akka
-    artifact: String,         // cats-core               | akka-http-experimental
+    groupId: String, // typelevel               | akka
+    artifactId: String, // cats-core               | akka-http-experimental
     version: SemanticVersion, // 0.6.0                   | 2.4.6
-    targets: ScalaTargets     // scalajs 0.6, scala 2.11 | scala 2.11
-  )
+    targets: ScalaTargets // scalajs 0.6, scala 2.11 | scala 2.11
+  ) extends GeneralReference {
+
+    def name: String = s"$groupId/$artifactId"
+    def httpUrl: String = s"/$groupId/$artifactId/$version"
+  }
 }
 
 // com.typesafe.akka - akka-http-experimental_2.11 - 2.4.6 | org.typelevel - cats-core_sjs0.6_2.11 - 0.6.0
 case class MavenReference(
   groupId: String,      // org.typelevel         | com.typesafe.akka
   artifactId: String,   // cats-core_sjs0.6_2.11 | akka-http-experimental_2.11
   version: String       // 0.6.0                 | 2.4.6
-)
+) extends GeneralReference {
+
+  def name: String = s"$groupId/$artifactId"
+  def httpUrl: String = s"http://mvnrepository.com/artifact/$groupId/$artifactId/$version"
```

Comment:
this link would had more info: `http://search.maven.org/#artifactdetails|com.typesafe.akka|akka-http-core_2.11|2.4.7|jar`